SSIZE_MAX undeclared for posix_filesystem

On my gentoo system using `gcc (Gentoo 7.2.0-r1 p1.1) 7.2.0` compilation fails for posic_filesystem.cc because SSIZE_MAX is undefined. This can be resolved by including `#include <limits.h>`.

Error:
```
In file included from /usr/include/dirent.h:233:0,
                 from /tmp/TileDB/tiledb/sm/filesystem/posix_filesystem.cc:40:
/tmp/TileDB/tiledb/sm/filesystem/posix_filesystem.cc: In function ‘tiledb::sm::Status tiledb::sm::posix::read(const string&, uint64_t, void*, uint64_t)’:
/tmp/TileDB/tiledb/sm/filesystem/posix_filesystem.cc:385:16: error: ‘LONG_MAX’ was not declared in this scope
   if (nbytes > SSIZE_MAX) {
```

--- a/tiledb/sm/filesystem/posix_filesystem.cc
+++ b/tiledb/sm/filesystem/posix_filesystem.cc
@@ -39,6 +39,8 @@
 
 #include <dirent.h>
 
+#include <limits.h>
+
 #include <ftw.h>
 
 #include <fstream>

